app/javascript/shared/components/tags.jsx

```diff
@@ -348,10 +280,33 @@ class Tags extends Component {
     }, 250);
   };
 
-  insertSpace(value, position) {
+  
+
+  insertSpace = (value, position) => {
     return `${value.slice(0, position)} ${value.slice(position, value.length)}`;
   }
 
+  insertTag(tag) {
+    const input = document.getElementById('tag-input');
+    const { maxTags, onInput } = this.props;
+    const range = this.getRangeBetweenCommas(input.value, input.selectionStart);
+    const insertingAtEnd = range[1] === input.value.length;
+    const maxTagsWillBeReached = this.selected.length === maxTags;
+    if (insertingAtEnd && !maxTagsWillBeReached) {
+      tag += ', ';
+    }
+
+    // Insert new tag between commas if there are any.
+    const newInput =
+      input.value.slice(0, range[0]) +
+      tag +
+      input.value.slice(range[1], input.value.length);
+
+    onInput(newInput);
+    this.resetSearchResults();
+    this.clearSelectedSearchResult();
+  }
+
   search(query) {
     if (query === '') {
       return new Promise(resolve => {
```

Comment:
there's a `'search query'` string on line 314 which can be removed

Reply:
Changes Updated @rhymes